Add some developer documentation for is_known. (#5199)
oscar-system/Oscar.jl: docs/src/DeveloperDocumentation/known_properties.md

```@meta
CurrentModule = Oscar
CollapsedDocStrings = true
DocTestSetup = Oscar.doctestsetup()
```
# Known properties

> Many high-level objects in Oscar such as, e.g. ideals in polynomial rings, algebraic schemes, etc. have properties which can be computed and eventually cached. For example, `krull_dim(I::MPolyIdeal)` returns the Krull dimension of `I` and, if not already present, caches the result in order to avoid a second computation of this property in the future. 

Reviewer: ```suggestion
Many high-level objects in Oscar such as ideals in polynomial rings, algebraic schemes, etc. have properties which can be computed and possibly be cached. For example, `krull_dim(I::MPolyIdeal)` returns the Krull dimension of `I` and, if not already present, caches the result in order to avoid a second computation of this property in the future. 
```
Not sure what the `eventually` was supposed to say.

In various settings it is then convenient to assess whether or not a certain property is already known, or whether calling it would trigger a non-trivial computation. As an example, think of a user-facing method to compute the normalization of the zero locus of an ideal `I` as above. There are specialized routines for curves which are faster than the generic ones for affine schemes of arbitrary dimension. Thus if the dimension of `I` was already known to be one, one 
could deflect for such a special routine directly. Yet, computation of an ideal's Krull dimension involves a Gröbner basis computation which, if not already done, may clog the main process for an indefinite amount of time. This risk of blockage only for reasons of algorithm selection might be considered too high: The user ``only'' wants a normalization, but Oscar gets stuck on gathering information on which implementation to choose from. For the user's convenience, one could then write
```julia
  function my_normalization(I::MPolyIdeal)

>     if is_known(krull_dim, I) && is_one(dim(I))

Reviewer: ```suggestion
    if is_known(krull_dim, I) && is_one(krull_dim(I))
```

      return special_and_fast_normalization_for_curves(I)
    else
      return generic_but_slow_normalization(I)
    end
  end
```
using methods of `is_known` for this setup.
```@docs
    is_known
```
Some further remarks:

> 

Reviewer: ```suggestion

  * The `is_known` methods have to be implemented manually, and do not automatically exist for all arguments.
```
